src/image-viewer.js

```diff
@@ -0,0 +1,650 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This class is responsible providing all operations necessary for viewing
+ * an image, such as full-bleed display, zoom and pan, etc.
+ * // TODO (cathyzhu): figure out what package visibility means
+ * @package  Visible for testing only!
+ */
+import {Animation} from './animation';
+import {Gestures} from './gesture';
+import {
+  DoubletapRecognizer,
+  SwipeXYRecognizer,
+  TapzoomRecognizer,
+} from './gesture-recognizers';
+import {bezierCurve} from './curve';
+import {isLoaded} from './event-helper';
+import {Services} from './services';
+import {continueMotion} from './motion';
+import {srcsetFromElement} from './srcset';
+import {
+  layoutRectFromDomRect,
+  layoutRectLtwh,
+  moveLayoutRect,
+} from './layout-rect';
+import * as st from './style';
+import * as tr from './transition';
+
+/** @private @const {!./curve.CurveDef} */
+const PAN_ZOOM_CURVE_ = bezierCurve(0.4, 0, 0.2, 1.4);
+
+export class ImageViewer {
```

Comment:
Some of the changes here seem to have broken the existing `amp-image-lightbox` (move up/down does not close it anymore and pinch to zoom seems to behave differently).

Given `amp-image-lightbox` is in production and we have to be careful with it (but lightbox 2.0 is not and it is okay to break it) and given there will be more changes to this as you work on lightbox 2.0, to eliminate the need to constantly QA `amp-image-lightbox` while lightbox 2.0 is in development, I recommend keeping the old copy of this in `amp-image-lightbox` and switch to the new implementation a bit later when lightbox 2.0 is more solid. WDYT?

Reply:
Reverted all changes to `amp-image-lightbox` and `test-amp-image-lightbox`.